docs/madr/decisions/070-resource-identifier.md

```diff
@@ -0,0 +1,363 @@
+| FilterChain | when `<match-value> == <protocol>` -> `meshpassthrough_<protocol>_<port>`<br>when `<match-value> != <protocol>` -> `meshpassthrough_<protocol>_<match-value>_<port>`<br>when port is 0 we put `*`, i.e. `meshpassthrough_http_*`<br>`<match-value>` is defined above                                                                                                                                                               | –                    |
+
+##### Internal
+
+|                  | Name                                                                                                                                                                 | Correlated Resources |
+|------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------|
+| Listener         | `kuma:dns`<br>`kuma:envoy:admin`<br>`kuma:metrics:prometheus`<br>`_kuma:metrics:opentelemetry:<backendName>`                                                         | –                    |
+| VirtualHost      | listener's name                                                                                                                                                      | –                    |
+| Internal Cluster | `kuma:readiness`<br>`kuma:envoy:admin`<br>`kuma:metrics:hijacker`<br>`_kuma:metrics:opentelemetry:<backendName>`<br>`tracing:`<br>`access_log_sink`<br>`ads_cluster` | –                    |
+
+#### Envoy stats
+
+There are 2 fields to specify alternative name for emitting stats – `stat_prefix`, `alt_stat_name`.
+Field `alt_stat_name` is used only for Cluster, and it's automatically converts all `:` to `_` by Envoy when emitting statistics.
+Field `stat_prefix` is used for HTTPConnectionManager, TCPProxy, KafkaBroker (network filter) and RBAC (network filter).
+Unlike `alt_stat_name` the field `stat_prefix` doesn't have any restrictions from Envoy.
+Kuma sanitizes both fields using `SanitizeMetric` function before setting them on Envoy's resource:
+
+```go
+var illegalChars = regexp.MustCompile(`[^a-zA-Z_\-0-9]`)
+
+// We need to sanitize metrics in order to  not break statsd and prometheus format.
+// StatsD only allow [a-zA-Z_\-0-9.] characters, everything else is removed
+// Extra dots breaks many regexes that converts statsd metric to prometheus one with tags
+func SanitizeMetric(metric string) string {
+	return illegalChars.ReplaceAllString(metric, "_")
+}
+```
+
+I can't find explanation why `alt_stat_name` and `stat_prefix` are treated differently by Envoy. 
+When Envoy exports metrics in Prometheus format resource name is always emitted as label value, it's never part of the metric name, i.e:
+```
+envoy_cluster_upstream_cx_total{envoy_cluster_name="localhost_5000"} 0
+```
+so there is no reason to replace `:` with `_`.
```

Comment:
In the past there were stuff in the metric name, I remember we even had some code to change things around. My is that this is the main reason no?

Also when not using prom format doesn't it actually use `:` as a separator see:

<img width="770" alt="image" src="https://github.com/user-attachments/assets/a47f9fe9-d36b-401a-9a6d-cf85e4c13237" />

Reply:
> In the past there were stuff in the metric name, I remember we even had some code to change things around. My is that this is the main reason no?

Yeah maybe, but then it's still unclear, why don't they replace `:` in `stat_prefix`? IMO it's just odd that `stat_prefix` and `alt_stat_name` don't have consistent behavior.

> Also when not using prom format doesn't it actually use : as a separator see:

Even though envoy uses `:` as a separator in `/clusters` endpoint, it still takes `cluster.name`, not `cluster.alt_stat_name`:

<img width="384" alt="image" src="https://github.com/user-attachments/assets/2aca6b75-82aa-4213-82cd-010aae9c4f7e" />

The cluster in the example named `kuma:readiness`